### PR TITLE
Add post validations

### PR DESCRIPTION
**Pull request by nicolaguidi**
I also added a jquery plugin to validate data before submitting the post. For some reason I couldn't get the body of a new post to work.

### CODE REVIEW

File: test/models/post_test.rb [hunk 1]
```diff
@@ -21,4 +21,16 @@ class PostTest < ActiveSupport::TestCase
     "own idea.", post.og_description
     assert_equal "http://www.thefirehoseproject.com/assets/meta-tag-coder.png", post.og_image
   end
+
```
Comment (kenmazaika) [inline test/models/post_test.rb:24]: I personally don't think validations are worth testing, but if you want to you might want to look into `shoulda-matchers` gem.

https://github.com/thoughtbot/shoulda-matchers

Allows you to write tests like:

```
  should validate_presence_of(:name)
```

I suggest either deleting these tests, or adding the shoulda-matchers gem to write code in a clearer way.

**Comment by nicolaguidi** [comment]
I'm deleting those tests, I believe is overkill to add a gem for a couple of tests.
Also: I'd like to know what is your opinion on what to test and what not to; I still struggle to make these decisions.

**Comment by kenmazaika** [comment]
I personally think it's overkill to test specific validations (although testing the case in controllers where things are invalid makes sense).

One of my coworkers was adamant about testing everything, and he was a solid developer.  Different developers feel differently on the topic, which is probably why it's a bit hard to know what opinion to have.

I'd say just remove the tests.

**Comment by kenmazaika** [comment]
@nicolaguidi can you resolve the merge conflicts?  After that we can merge this in.

**Comment by nicolaguidi** [comment]
Solved the merge conflicts. I couldn't run the migration on my machine, not sure this can affect the merge commit.

**Comment by kenmazaika** [comment]
Do the migrations give you errors @nicolaguidi?  If so, we should probably open a ticket to fix those!

**Comment by nicolaguidi** [comment]
It's probably related to my local postgres installation, this is the error I'm getting:

```
could not connect to server: Connection refused
    Is the server running on host "localhost" (::1) and accepting
    TCP/IP connections on port 5432?
...
```

**Comment by kenmazaika** [comment]
Hmmm.  Looks like it is.  I might suggest reworking you vagrant setup.

```
vagrant destroy
vagrant up
```

And then setting up your SSH keys.

Then your postgres should be working (assuming you're using vagrant).

**Pull request merged**